### PR TITLE
uboot_auto_patch: handle more complex dependency strings …

### PR DESCRIPTION
**Pull request by joergho**
…in add_kconfig_option_with_depends script

The change makes the script capable of handling more complex depends-strings which
contain or-operators ("||") where the first alternative is used, parentheses, and
and-operators ("&&").

The "boolean" library from Sebastian Kraemer is used for evaluating the logical expressions.
It was copied from https://github.com/bastikr/boolean.py/blob/master/boolean/boolean.py
(dd4bc01) as there is no python3-boolean recipe available for the current openembedded releases.

This makes the script compatible with U-Boot 2019.10.

U-Boot 2019.10 has the following ENV_OFFSET dependencies:
    depends on (!ENV_IS_IN_UBI && !ENV_IS_NOWHERE) || ARCH_STM32MP
(see https://gitlab.denx.de/u-boot/u-boot/blob/v2019.10/env/Kconfig#L469)

In comparison, U-Boot 2019.01 has the following ENV_OFFSET dependencies:
    depends on !ENV_IS_IN_UBI
    depends on !ENV_IS_NOWHERE
(see https://gitlab.denx.de/u-boot/u-boot/blob/v2019.01/env/Kconfig#L436)

Changelog: Update uboot_auto_patch script to be compatible with U-Boot 2019.10

Signed-off-by: Joerg Hofrichter <joerg.hofrichter@ni.com>

### CODE REVIEW

**Comment by kacf** [comment]
@joergho: I'm seeing some unrelated errors probably due to the branch being out of date. Could you please rebase your branch on latest master and force push?

**Comment by joergho** [comment]
@kacf Done: 6796dc8

File: meta-mender-core/recipes-bsp/u-boot/u-boot-mender-common.inc [hunk 1]
```diff
@@ -290,6 +290,7 @@ SRC_URI_append_mender-uboot = "${@bb.utils.contains('MENDER_UBOOT_AUTO_CONFIGURE
                                                     ' file://uboot_auto_configure.sh \
                                                       file://uboot_auto_patch.sh \
                                                       file://add_kconfig_option_with_depends.py', \
+                                                      file://boolean.py', \
```
Comment (kacf) [inline meta-mender-core/recipes-bsp/u-boot/u-boot-mender-common.inc]: You've got a typo here: Only one of the lines should have the terminating `'`.

Comment (joergho) [reply meta-mender-core/recipes-bsp/u-boot/u-boot-mender-common.inc]: Oh sorry for overlooking it -> b070331

Comment (kacf) [reply meta-mender-core/recipes-bsp/u-boot/u-boot-mender-common.inc]: That was quick, thanks!

**Comment by kacf** [comment]
Thanks, @joergho!

**Pull request merged**